cmd/common-main.go

```diff
@@ -108,19 +113,10 @@ func init() {
 
 	globalIsCICD = env.Get("MINIO_CI_CD", "") != "" || env.Get("CI", "") != ""
 
-	containers := IsKubernetes() || IsDocker() || IsBOSH() || IsDCOS() || IsPCFTile()
-
-	// Call to refresh will refresh names in cache. If you pass true, it will also
-	// remove cached names not looked up since the last call to Refresh. It is a good idea
-	// to call this method on a regular interval.
+	// Call to refresh will refresh names in cache.
 	go func() {
-		var t *time.Ticker
-		if containers {
-			// k8s DNS TTL is 30s (Attempt a refresh only after)
-			t = time.NewTicker(30 * time.Second)
-		} else {
-			t = time.NewTicker(10 * time.Minute)
-		}
+		// baremetal setups set DNS refresh window to 30 minutes.
+		t := time.NewTicker(30 * time.Minute)
```

Comment:
Isn't 30 minutes very high ? especially with persist errors.. this means an error will be persisted for 30 minutes

Reply:
For baremetal setups this is not a real problem @vadmeste

Reply:
Remember refresh() doesn't wait for 30mins to refresh. The issue was that the failure to refresh invalidated previous successful result @vadmeste

Reply:
> For baremetal setups this is not a real problem @vadmeste - since the initial lookup always succeeds once.

The issue is when the first lookup fails, the library will persist it until someone refreshes it. This means sometimes when you start a cluster a node can take 30 minutes to start because of this. I took a look at the library code, the library doesn't try to look up again if it never had a success before.